rpm/0071-sailfishos-webrtc-Implement-video-capture-module.-JB.patch

```diff
@@ -0,0 +1,797 @@
+-    return;
++  switch (aFrame.rotation()) {
++    case webrtc::kVideoRotation_90:
++      rotationMode = libyuv::kRotate90;
++      std::swap(width, height);
++      break;
++    case webrtc::kVideoRotation_270:
++      rotationMode = libyuv::kRotate270;
++      std::swap(width, height);
++      break;
++    case webrtc::kVideoRotation_180:
++      rotationMode = libyuv::kRotate180;
++      break;
++    case webrtc::kVideoRotation_0:
++    default:
++      rotationMode = libyuv::kRotate0;
++      break;
+   }
+ 
+-  // Copy each plane
+-  size_t offset = 0;
+-  size_t size;
+-  auto height = aFrame.video_frame_buffer()->height();
+-  size = height * aFrame.video_frame_buffer()->StrideY();
+-  memcpy(&aDestBuffer[offset], aFrame.video_frame_buffer()->DataY(), size);
+-  offset += size;
+-  size = ((height + 1) / 2) * aFrame.video_frame_buffer()->StrideU();
+-  memcpy(&aDestBuffer[offset], aFrame.video_frame_buffer()->DataU(), size);
+-  offset += size;
+-  size = ((height + 1) / 2) * aFrame.video_frame_buffer()->StrideV();
+-  memcpy(&aDestBuffer[offset], aFrame.video_frame_buffer()->DataV(), size);
++  int strideY = width;
++  int strideUV = (width + 1) / 2;
++  off_t offsetY = 0;
++  off_t offsetU = height * strideY;
++  off_t offsetV = offsetU + ((height + 1) / 2) * strideUV;
++
++  libyuv::I420Rotate(src->DataY(), src->StrideY(),
++                     src->DataU(), src->StrideU(),
++                     src->DataV(), src->StrideV(),
++                     &aDestBuffer[offsetY], strideY,
++                     &aDestBuffer[offsetU], strideUV,
++                     &aDestBuffer[offsetV], strideUV,
++                     src->width(), src->height(),
++                     rotationMode);
+ }
+ 
+ void VideoFrameUtils::CopyVideoFrameBuffers(
+diff --git a/dom/media/systemservices/moz.build b/dom/media/systemservices/moz.build
+index cafe81bf84d0..0fbe06b497ba 100644
+--- a/dom/media/systemservices/moz.build
++++ b/dom/media/systemservices/moz.build
+@@ -20,6 +20,7 @@ if CONFIG['MOZ_WEBRTC']:
+         'VideoFrameUtils.cpp'
+     ]
+     LOCAL_INCLUDES += [
++        '/media/libyuv/libyuv/include/',
+         '/media/webrtc/signaling',
+         '/media/webrtc/trunk',
+     ]
+diff --git a/media/webrtc/trunk/webrtc/modules/video_capture/BUILD.gn b/media/webrtc/trunk/webrtc/modules/video_capture/BUILD.gn
+index e4984238bac3..c2f10a7a7e87 100644
+--- a/media/webrtc/trunk/webrtc/modules/video_capture/BUILD.gn
++++ b/media/webrtc/trunk/webrtc/modules/video_capture/BUILD.gn
+@@ -98,10 +98,12 @@ if (!build_with_chromium) {
+ 
+     if (is_linux || is_bsd) {
+       sources = [
+-        "linux/device_info_linux.cc",
+-        "linux/device_info_linux.h",
+-        "linux/video_capture_linux.cc",
+-        "linux/video_capture_linux.h",
++        "gst/device_info_gst.cc",
++        "gst/device_info_gst.h",
++        "gst/video_capture_gst.cc",
++        "gst/video_capture_gst.h",
++        "gst/gst_camera.cpp",
++        "gst/gst_camera.h",
```

Comment:
Gstreamer's not used for anything now right? Should these be removed or replaced with sfos equivalents?